attributed_text/src/attributed_text.rs

```diff
@@ -143,32 +127,76 @@ mod tests {
         let t = "Hello!";
         let mut at = AttributedText::new(t);
 
-        assert_eq!(at.apply_attribute(0..3, TestAttribute::Keep), Ok(()));
-        assert_eq!(at.apply_attribute(0..6, TestAttribute::Keep), Ok(()));
-        assert_eq!(
-            at.apply_attribute(4..3, TestAttribute::Keep),
-            Err(ApplyAttributeError::InvalidRange)
-        );
-        assert_eq!(
-            at.apply_attribute(0..7, TestAttribute::Keep),
-            Err(ApplyAttributeError::InvalidBounds)
-        );
-        assert_eq!(
-            at.apply_attribute(7..8, TestAttribute::Keep),
-            Err(ApplyAttributeError::InvalidBounds)
-        );
+        assert!(at.apply_attribute(0..3, TestAttribute::Keep).is_ok());
+        assert!(at.apply_attribute(0..6, TestAttribute::Keep).is_ok());
+        match at.apply_attribute(4..3, TestAttribute::Keep) {
+            Err(e) => {
+                assert_eq!(e.kind(), ErrorKind::InvalidRange);
+                let msg = format!("{}", e);
+                assert!(msg.contains("4..3"));
+                assert!(msg.contains("start > end") || msg.contains("invalid range"));
```

Comment:
Do we expect to always keep saying that the range is invalid, and why it's invalid?

```suggestion
                assert!(msg.contains("start > end"));
                assert!(msg.contains("invalid range"));
```